src/CommunitiesList.cpp

```diff
@@ -56,6 +64,59 @@ CommunitiesList::setCommunities(const mtx::responses::JoinedGroups &response)
 
         communities_["world"]->setPressedState(true);
         emit communityChanged("world");
+        sortEntries();
+}
+
+void
+CommunitiesList::syncTags(const std::map<QString, RoomInfo> &info)
+{
+        for (const auto &room : info)
+                setTagsForRoom(room.first, room.second.tags);
+        sortEntries();
+}
+
+void
+CommunitiesList::setTagsForRoom(const QString &room_id, const std::vector<std::string> &tags)
+{
+        // create missing tag if any
+        for (const auto &tag : tags) {
+                // filter out tags we should ignore according to the spec
+                // https://matrix.org/docs/spec/client_server/r0.4.0.html#id154
+                // nheko currently does not make use of internal tags
+                // so we ignore any tag containig a `.` (which would indicate a tag
+                // in the form `tld.domain.*`) except for `m.*` and `u.*`.
+                if (tag.find(".") != ::std::string::npos && tag.compare(0, 2, "m.") &&
+                    tag.compare(0, 2, "m."))
+                        continue;
+                QString name = QString("tag:") + QString::fromStdString(tag);
+                if (!communityExists(name)) {
+                        addCommunity(std::string("tag:") + tag);
+                }
+        }
+        // update membership of the room for all tags
+        auto it = communities_.begin();
+        while (it != communities_.end()) {
```

Comment:
Maybe I'm missing something but can't this be translated into a standard for-range loop?

Reply:
I don't know how to do the `it = communities_.erase(it);` (at like 115) with a standard for-range loop, is that possible?

Reply:
You're right, I missed that part. You would need the iterator.